geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala

```diff
@@ -164,13 +167,53 @@ class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with L
    *
    * Our delegated AccumuloInputFormat creates a split for each range - because we set a lot of ranges in
    * geomesa, that creates too many mappers. Instead, we try to group the ranges by tservers. We use the
-   * number of shards in the schema as a proxy for number of tservers.
+   * location assignment of the tablets to tservers to determine the number of splits returned.
    */
   override def getSplits(context: JobContext): java.util.List[InputSplit] = {
     init(context.getConfiguration)
     val accumuloSplits = delegate.getSplits(context)
-    logger.debug(s"Got ${accumuloSplits.length} splits")
-    accumuloSplits
+    // Get the appropriate number of mapper splits using the following priority
+    // 1. Get splits from AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS (geomesa.mapreduce.splits.max)
+    // 2. Get splits from #tserver locations * AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER (geomesa.mapreduce.splits.tserver.max)
+    // 3. Get splits from AccumuloInputFormat.getSplits(context)
+    val grpSplitsMax: Option[Int] = AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS.option.flatMap { prop =>
+      try {
+        Some(prop.toInt).filter(_ > 0)
+      } catch {
+        case e: java.lang.NumberFormatException =>
+          throw new IllegalArgumentException(s"Unable to parse geomesa.mapreduce.splits.max = $prop contains an invalid Int.", e)
+      }
+    }
+
+    lazy val grpSplitsPerTServer: Option[Int] =
+      AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER.option match {
+        case Some(desiredSplits) =>
+          val numLocations = accumuloSplits.flatMap(_.getLocations).toArray.distinct.length
+          val splitsPerTServer = try { desiredSplits.toInt } catch {
+            case e: java.lang.NumberFormatException =>
+              logger.warn(s"Unable to parse geomesa.mapreduce.splits.tserver.max = " + s"$desiredSplits is not a valid Int.")
+              1 // Identity, don't split locations
+          }
+          if (numLocations > 0 && splitsPerTServer > 0) Some(numLocations * splitsPerTServer) else None
+        case None => None
+      }
+
+    grpSplitsMax.orElse(grpSplitsPerTServer) match {
```

Comment:
I think the idea was to specify number of mappers per shard instead of always defaulting to 2. Along with that, it would be good to use something besides shards (which we don't even have any more since they were for the geohash index) to determine the number of tservers.